File: openfinalfinal/u202120418/u202120418/src/main/java/org/hign/platform/u202120418/personnel/interfaces/rest/transform/ExaminerResourceFromEntityAssembler.java
package org.hign.platform.u202120418.personnel.interfaces.rest.transform;/**
 * @author Fiorella Jarama Peñaloza
 * @version 1.0
 */

import org.hign.platform.u202120418.personnel.domain.model.aggregates.Examiner;
import org.hign.platform.u202120418.personnel.domain.model.commands.CreateExaminerCommand;
import org.hign.platform.u202120418.personnel.interfaces.rest.resources.CreateExaminerResource;
import org.hign.platform.u202120418.personnel.interfaces.rest.resources.ExaminerResource;

/**
 * ExaminerResourceFromEntityAssembler - Assembler for the command to create an examiner from a resource
 *
 * @author Fiorella Jarama Peñaloza
 */
public class ExaminerResourceFromEntityAssembler {
    /**
     * Transforms a CreateExaminerResource into a CreateExaminerCommand.
     * @param entity the CreateExaminerResource.
     * @return the CreateExaminerCommand.
     */
    public static ExaminerResource toResourceFromEntity(Examiner entity){
        return new ExaminerResource(
                entity.getId(),
                entity.getFirstName(),
                entity.getLastName(),
                entity.getNationalProviderIdentifier());
    }
}

File: web/HIGN/HIGN/Personnel/Interfaces/REST/Transform/ExaminerResourceFromEntityAssembler.cs
using HIGN.Personnel.Domain.Model.Aggregates;
using HIGN.Personnel.Interfaces.REST.Resources;

namespace HIGN.Personnel.Interfaces.REST.Transform;

public class ExaminerResourceFromEntityAssembler
{
    public static ExaminerResource ToResourceFromEntity(Examiner examiner)
    {
        return new ExaminerResource(examiner.Id, examiner.FirstName, examiner.LastName,
            examiner.NationalProviderIdentifier);
    }
}